Student/Student/Profile/ProfilePresenter.swift

```diff
@@ -0,0 +1,155 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import CanvasCore
+import Core
+
+public class ProfilePresenter: ProfilePresenterProtocol {
+    let env: AppEnvironment
+    public weak var view: ProfileViewProtocol?
+
+    #if DEBUG
+    var showDevMenu = true
+    #else
+    var showDevMenu = UserDefaults.standard.bool(forKey: "showDevMenu")
+    #endif
+
+    public lazy var helpLinks = env.subscribe(GetAccountHelpLinks(for: .student)) { [weak self] in
+        self?.view?.reload()
+    }
+
+    lazy var settings = env.subscribe(GetUserSettings(userID: "self")) { [weak self] in
+        self?.view?.reload()
+    }
+
+    lazy var tools = env.subscribe(GetGlobalNavExternalPlacements()) { [weak self] in
+        self?.view?.reload()
+    }
+
+    lazy var permissions = env.subscribe(GetContextPermissions(context: ContextModel(.account, id: "self"), permissions: [.becomeUser])) { [weak self] in
```

Comment:
Now that we are using this in Student, I think it would be best to update it to not refresh the token since it's almost always going to be a 401.. You would do that by implementing `makeRequest` and passing `refreshToken: false` to the `api.makeRequest` call.